libs/agno/tests/unit/playground/test_image_support_file_upload.py

```diff
@@ -0,0 +1,218 @@
+"""
+Unit tests for async_router file upload functionality.
+"""
+
+import io
+from typing import List
+from unittest.mock import AsyncMock, Mock
+
+import pytest
+from fastapi import APIRouter, File, Form, HTTPException, UploadFile
+from fastapi.testclient import TestClient
+
+
+# Define an EmptyFile class that simulates an empty file without raising on read.
+class EmptyFile:
+    def __init__(self, content: bytes):
+        self.content = content
+        self._read_called = False
+
+    def read(self, n=-1):
+        # Always return empty bytes. The endpoint will then raise a ValueError.
+        if not self._read_called:
+            self._read_called = True
+            return b""
+        return b""
+
+    def seek(self, offset, whence=0):
+        pass
+
+
+# --- Fixtures ---
+
+
+@pytest.fixture
+def mock_agent():
+    """Creates a mock agent with minimal attributes required by the router."""
+    mock = Mock()
+    mock.agent_id = "test_agent_id"
+    mock.name = "Test Agent"
+    # Assume deep_copy returns self.
+    mock.deep_copy = lambda update=None: mock
+    # Dummy asynchronous run method.
+    mock.arun = AsyncMock(return_value=iter([{"dummy": "response"}]))
+    # Set knowledge to None by default
+    mock.knowledge = None
+    return mock
+
+
+@pytest.fixture
+def test_app(mock_agent):
```

Comment:
This doesn't really help, because you aren't really testing anything... since you are replacing it.